airflow/providers/google/leveldb/hooks/leveldb.py

```diff
@@ -17,11 +17,24 @@
 """Hook for Level DB"""
 from typing import List, Optional
 
-import plyvel
-from plyvel import DB
-
-from airflow.exceptions import AirflowException
-from airflow.hooks.base import BaseHook
+try:
+    import plyvel
+    from plyvel import DB
+
+    from airflow.exceptions import AirflowException
+    from airflow.hooks.base import BaseHook
+
+except ImportError as e:
+    # Plyvel is an optional feature and if imports are missing, it should be silently ignored
+    # As of Airflow 2.3  and above the operator can throw OptionalProviderFeatureException
+    try:
+        from airflow.exceptions import AirflowOptionalProviderFeatureException
+    except ImportError:
+        # However, in order to keep backwards-compatibility with Airflow 2.1 and 2.2, if the
+        # 2.3 exception cannot be imported, the original ImportError should be raised.
+        # This try/except can be removed when the provider depends on Airflow >= 2.3.0
+        raise e
```

Comment:
Better to do `raise e from None` here to avoid the nested `ImportError` confusing the traceback.

Reply:
Ah nice. TIL. I knew just `raise` removes exception context, but I could not use it due to exception-in-exception. Thanks!